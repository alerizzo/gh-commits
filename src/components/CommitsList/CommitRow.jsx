import React from 'react';
import { Skeleton } from 'components';
//import Gravatar from 'react-gravatar';

const moment = require('moment');

const CommitRow = ({ commit }) => {
  return (
    <div className="columns CommitRow">
      <div className="column is-2" title={commit.author.email}>
        <figure className="image is-28x28">
          <img className="is-rounded" src={commit.author.avatarUrl} alt={commit.author.name} />
          {/* author.avatarUrl is always present  */}
          {/* <Gravatar
            email={commit.author.email}
            size={28}
            rating="pg"
            default={commit.author.avatarUrl}
            className="is-rounded"
          /> */}
        </figure>
        <strong>{commit.author.name}</strong>
      </div>
      <div className="column is-2">
        {commit.abbreviatedOid}
        {commit.parents.totalCount > 1 && <span className="tag is-primary is-light">Merge</span>}
      </div>
      <div className="column is-5">{commit.messageHeadline}</div>
      <div className="column is-3">
        <time dateTime={commit.commitedDate}>{moment(commit.committedDate).fromNow()}</time>
      </div>
    </div>
  );
};

CommitRow.Header = () => {
  return (
    <div className="columns CommitRow CommitRow_Header">
      <div className="column is-2">Author</div>
      <div className="column is-2">Commit</div>
      <div className="column is-5">Message</div>
      <div className="column is-3">Created</div>
    </div>
  );
};

CommitRow.Skeleton = ({ loading }) => {
  return (
    <div className="columns CommitRow">
      <div className="column is-2">
        <Skeleton.Circle loading={loading} size="28px" />
        <Skeleton.Text loading={loading} size="60%" />
      </div>
      <div className="column is-2">
        <Skeleton.Text loading={loading} size="50%" />
      </div>
      <div className="column is-5">
        <Skeleton.Text loading={loading} size="70%" />
      </div>
      <div className="column is-3">
        <Skeleton.Text loading={loading} size="50%" />
      </div>
    </div>
  );
};

export default CommitRow;
