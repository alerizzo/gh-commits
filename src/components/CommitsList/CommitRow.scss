@import '../../scss/variables';

.CommitRow {
  border-bottom: solid 1px $border;

  &.CommitRow_Header {
    border-bottom: solid 2px $border;
  }

  .column {
    padding: 0.55rem 0.75rem;
  }

  figure {
    flex-shrink: 0;
  }
}
