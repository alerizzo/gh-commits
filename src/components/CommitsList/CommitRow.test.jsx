import React from 'react';
import renderer from 'react-test-renderer';

import CommitRow from './CommitRow';

describe('<CommitRow />', () => {
  const defaultProps = {};
  const wrapper = renderer.create(<CommitRow {...defaultProps} />);

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<CommitRow commit={ ... } />', () => {
  const defaultProps = {
    commit: {
      id: 'MDY6Q29tbWl0MTU3NjIxOTYzOjE3MmJkYjc4MDU5NDA0ZjE5MzA1OWEyOGJkMzkzZDA4NmM0NjZhOTE=',
      abbreviatedOid: '172bdb7',
      messageHeadline: 'Sample message headline',
      committedDate: '2019-10-08T17:00:00Z',
      author: {
        name: 'Tyler Durden',
        email: 'tdurden@paperstsoap.co',
        avatarUrl: 'https://avatars0.githubusercontent.com/u/92839?v=4',
      },
      parents: {
        totalCount: 2,
      },
    },
  };

  const wrapper = renderer.create(<CommitRow {...defaultProps} />);

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<CommitRow.Header />', () => {
  const defaultProps = {};
  const wrapper = renderer.create(<CommitRow.Header {...defaultProps} />);

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<CommitRow.Skeleton />', () => {
  const defaultProps = {};
  const wrapper = renderer.create(<CommitRow.Skeleton {...defaultProps} />);

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
