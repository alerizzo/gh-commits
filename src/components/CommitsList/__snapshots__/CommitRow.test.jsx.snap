// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommitRow /> render 1`] = `
<div
  className="columns CommitRow"
>
  <div
    className="column is-2"
  >
    <div
      className="Skeleton Skeleton_Circle "
      style={
        Object {
          "height": "28px",
          "width": "28px",
        }
      }
    />
    <div
      className="Skeleton Skeleton_Text "
      style={
        Object {
          "width": "60%",
        }
      }
    />
  </div>
  <div
    className="column is-2"
  >
    <div
      className="Skeleton Skeleton_Text "
      style={
        Object {
          "width": "50%",
        }
      }
    />
  </div>
  <div
    className="column is-5"
  >
    <div
      className="Skeleton Skeleton_Text "
      style={
        Object {
          "width": "70%",
        }
      }
    />
  </div>
  <div
    className="column is-3"
  >
    <div
      className="Skeleton Skeleton_Text "
      style={
        Object {
          "width": "50%",
        }
      }
    />
  </div>
</div>
`;

exports[`<CommitRow commit={ ... } /> render 1`] = `
<div
  className="columns CommitRow"
>
  <div
    className="column is-2"
    title="tdurden@paperstsoap.co"
  >
    <figure
      className="image is-28x28"
    >
      <img
        alt="Tyler Durden"
        className="is-rounded"
        src="https://avatars0.githubusercontent.com/u/92839?v=4"
      />
    </figure>
    <strong>
      Tyler Durden
    </strong>
  </div>
  <div
    className="column is-2"
  >
    172bdb7
    <span
      className="tag is-primary is-light"
    >
      Merge
    </span>
  </div>
  <div
    className="column is-5"
  >
    Sample message headline
  </div>
  <div
    className="column is-3"
  >
    <time>
      a month ago
    </time>
  </div>
</div>
`;

exports[`<CommitRow.Header /> render 1`] = `
<div
  className="columns CommitRow CommitRow_Header"
>
  <div
    className="column is-2"
  >
    Author
  </div>
  <div
    className="column is-2"
  >
    Commit
  </div>
  <div
    className="column is-5"
  >
    Message
  </div>
  <div
    className="column is-3"
  >
    Created
  </div>
</div>
`;

exports[`<CommitRow.Skeleton /> render 1`] = `
<div
  className="columns CommitRow"
>
  <div
    className="column is-2"
  >
    <div
      className="Skeleton Skeleton_Circle "
      style={
        Object {
          "height": "28px",
          "width": "28px",
        }
      }
    />
    <div
      className="Skeleton Skeleton_Text "
      style={
        Object {
          "width": "60%",
        }
      }
    />
  </div>
  <div
    className="column is-2"
  >
    <div
      className="Skeleton Skeleton_Text "
      style={
        Object {
          "width": "50%",
        }
      }
    />
  </div>
  <div
    className="column is-5"
  >
    <div
      className="Skeleton Skeleton_Text "
      style={
        Object {
          "width": "70%",
        }
      }
    />
  </div>
  <div
    className="column is-3"
  >
    <div
      className="Skeleton Skeleton_Text "
      style={
        Object {
          "width": "50%",
        }
      }
    />
  </div>
</div>
`;
